core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -907,11 +918,21 @@ class ReplicaManager(val config: KafkaConfig,
       }
     }
 
-    // Wrap the given callback function with another function that will update the HW for the remote follower
-    def maybeUpdateHwAndSendResponse(fetchPartitionData: Seq[(TopicPartition, FetchPartitionData)]): Unit = {
+    // Wrap the given callback function with another function that will update the HW and partition reassignment lag
+    // for the remote follower
+    def updateReplicaAndSendResponse(fetchPartitionData: Seq[(TopicPartition, FetchPartitionData)]): Unit = {
       if (isFromFollower) {
         fetchPartitionData.foreach {
-          case (tp, partitionData) => updateFollowerHighWatermark(tp, replicaId, partitionData.highWatermark)
+          case (tp, partitionData) =>
+            val replicaOpt = nonOfflinePartition(tp).flatMap(_.getReplica(replicaId))
+            replicaOpt match {
+              case Some(replica) =>
+                replica.updateLastSentHighWatermark(partitionData.highWatermark)
+                replica.updateReassignmentLag(partitionData.messageOffset, isAddingReplica(tp, replicaId))
```

Comment:
nit: any particular reason to rename the `maybeUpdateHwAndSendResponse` and not put the new `             replica.updateReassignmentLag(partitionData.messageOffset, isAddingReplica(tp, replicaId))` inside the `updateFollowerHighWatermark ` function?

I thought it abstracted things nicely

Reply:
Yea, I'll refactor this.